Tab theme configurable

DLFW opened the pull request:
Text style for tabs can be configured within
`theme.toml` with the dotted keys `tab.active` and `tab.inactive`.

The default style is the same as the former hard-coded style.

`theme.toml` has been restructured and comments have been added to give a short documentation on each style item.

The documentation (`theme.toml.md`) has been enhanced, and the excerpt of `theme.toml` with additional comments has been removed from there.

That way it should be easier for devs to keep docs and the default `theme.toml` in sync, and easier for users when the detail comments can be found in the configuration file itself.

The conversion from `AppStyle` to `tui::style::Style` is now provided as a function of `AppStyle` and a `From`-implementation for `Style` to avoid having the same factory code repeating all over.

kamiyaa commented on docs/configuration/theme.toml.md:
```diff
@@ -1,11 +1,14 @@
+This file is used to theme Joshuto.
 
-## Style
+As of now, all theming items are “text style” items where one can define the
```
Nit, I would use "" instead of “”. Unless you have a reason to use those quotes.

kamiyaa commented on docs/configuration/theme.toml.md:
```diff
@@ -15,81 +18,46 @@ bold = false
+[`theme.toml`](https://github.com/kamiyaa/joshuto/blob/main/config/theme.toml).
 
-System file types include:
+In general, `theme.toml` allows to specify the style of a few UI “widgets”,
```
Same nit here xd

kamiyaa reviewed (commented):
Looks good overall, just a few minor nits. 
Will merge once conflict is resolved as well as the nits are addressed :+1:

DLFW commented:
Rebased to main, fixed comments, and squashed.

The reason I had used “” is simply that these are the typographically correct quotation marks. However, I understand that this might be confusing and deviate from other places as most people use the ASCII "" quotation. I just have them in my muscle memory. I need to remember to use the "ASCII quotes" while writing prose. :smiling_face_with_tear:

kamiyaa commented:
Thanks!
